buildSrc/src/main/groovy/org/elasticsearch/gradle/test/RestIntegTestTask.groovy

```diff
@@ -55,7 +57,19 @@ public class RestIntegTestTask extends DefaultTask {
         super.dependsOn(runner)
         clusterInit = project.tasks.create(name: "${name}Cluster#init", dependsOn: project.testClasses)
         runner.dependsOn(clusterInit)
-        clusterConfig = project.extensions.create("${name}Cluster", ClusterConfiguration.class, project)
+        boolean usesTestclusters = project.plugins.hasPlugin(TestClustersPlugin.class)
```

Comment:
This is starting to become a bit of a mess, mainly because this logic should not live in a task constructor, but in the plugin.

It's also starting to become more and more like we should have two separate plugins that do this wiring, one that uses test clusters, and the legacy one. We could rip out common logic into a `BasePluginBuildPlugin` for DRY purposes. All this "if test clusters do this, if not do that" stuff is starting to get a little out of hand. 

I realize this is intended to be temporary but there is loads of code out there that folks intended to remove that has lived for years. Let's assume this hybrid approach and the migration to test clusters is going to be an extended effort.

Reply:
I do get your point on thinking it's temporary and then realizing it's not, but we have so many requirements on testclusters that it will be impossible to switch over. It's also my primary focus,  so unless I am hit by a bus, the replacement will happen. It's a good chance to take :)